portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -32,20 +37,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String urlToRedirectTo = "/";
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
+      String nickname = getUserNickname(userService.getCurrentUser().getUserId());
       String userEmail = userService.getCurrentUser().getEmail();
       String logoutUrl = userService.createLogoutURL(urlToRedirectTo);
-      response.getWriter().println(createJson("yes", logoutUrl, userEmail));
+      response.getWriter().println(createJson("yes", logoutUrl, userEmail, nickname));
     } else {
       String loginUrl = userService.createLoginURL(urlToRedirectTo);
-      response.getWriter().println(createJson("no", loginUrl, ""));
+      response.getWriter().println(createJson("no", loginUrl, "", ""));
     }
   }
 
-  private String createJson(String isLoggedIn, String url, String email){
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String nickname = request.getParameter("nickname");
+      String id = userService.getCurrentUser().getUserId();
+ 
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Entity entity = new Entity("UserInfo", id);
+      entity.setProperty("id", id);
+      entity.setProperty("nickname", nickname);
+      // The put() function automatically inserts new data or updates existing data based on ID
+      datastore.put(entity);
+      // Direct the user to main page
+      response.sendRedirect("/index.html");
+    }
+
+    
+  }
+
+  private String createJson(String isLoggedIn, String url, String email, String nickname){
```

Comment:
Optional:
Why not use Gson.toJson

You can create a minimal private sub class here